include/devicetree/gpio.h

```diff
@@ -190,6 +190,94 @@ extern "C" {
 #define DT_INST_GPIO_FLAGS(inst, gpio_pha) \
 	DT_INST_GPIO_FLAGS_BY_IDX(inst, gpio_pha, 0)
 
+/**
+ * @brief Create a struct initializer for a "gpio" phandle-array element
+ *
+ * The struct initializer will be of the form:
+ *
+ * { label, pin, flags },
+ *
+ * Example devicetree fragment:
+ *
+ *     gpio1: gpio@... {
+ *             label = "GPIO_1";
+ *     };
+ *
+ *     gpio2: gpio@... {
+ *             label = "GPIO_2";
+ *     };
+ *
+ *     n: node {
+ *             gpios = <&gpio1 10 20>, <&gpio2 30 40>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_GPIO_ELEM(DT_NODELABEL(n), gpios, 1)
+ *
+ * Will produce the following struct initializer:
+ *
+ *     { "GPIO_2", 30, 40 },
+ *
+ * The idx element is placed first so that the macro will be usable by
+ * UTIL_LISTIFY
+ *
+ * @param idx into the gpio_pha phandle-array
+ * @param node_id node identifier
+ * @param gpio_pha lowercase-and-underscores GPIO property with
+ *        type "phandle-array"
+ * @return a struct initializer for the give index "idx" into the
+ *         "phandle-array" gpio_pha
+ */
+#define DT_GPIO_ELEM(idx, node_id, gpio_pha) \
```

Comment:
What do you think about the name DT_GPIO_LPF_AT_IDX?

The other phandle array macros also take arguments in (node_id, gpio_pha, idx)  order.

Reply:
`DT_GPIO_CELLS_AT_IDX` would be more consistent with the underlying devicetree terminology and would be highlight the similarity with other phandle-valued aggregates.  I really find `_LPF_` to be obscure.  (Pins, Flags, what is L?)

Reply:
I think L is label

Reply:
OK, not obvious especially if label goes away.  I'd rather find a name that covers all phandle+specifiers so we can use it consistently for `DT_PWM_`, `DT_IOCHANNEL_`, ....  `CELLS` ignores the phandle, so maybe `DT_GPIO_PHCELLS_`?  Ideally when the phandle reference changes from a label to an identifier we wouldn't have to change the name, just the definition of the macro and of the structure into which it should be stored.

But really this PR is jumping to a partial solution for #23691 without addressing the other part: eliminating the custom `struct sx1276_dio`.  We do not need a global macro creating precedent and adding new names to convert lora/sx1276.c; that file can use a local define as I did in https://github.com/zephyrproject-rtos/zephyr/pull/23245#pullrequestreview-379023714